docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,300 @@
+and they perform one or more attempts to complete the key generation.
+
+Inactive members are removed from the DKG participants
+and the resulting wallet if the DKG is successful.
+
+If there is misbehaviour,
+the accusations will be resolved by governance
+as the protocol in `tss-lib` is not attributable.
+
+=== Parameter selection
+
+For tbtc2 wallets,
+we need to choose the group size _N_
+(how many members we select from the sortition pool)
+and the threshold _T_
+(how many members must cooperate to generate a threshold ECDSA signature).
+
+In addition, we need to choose the critical size _C_
+which is number of members required for a successful key generation.
+Some of the selected _N_ members may not be available for the key generation.
+To reduce sortition costs,
+we let some of the original _N_ be inactive
+as long as at least _C_ participants can be assembled to complete the DKG.
+After successful DKG completion,
+the created wallet will have a number of members
+that is between _C_ and _N_ (inclusive).
+The inactive members that did not participate in the successful DKG
+are removed from the wallet's members and are ineligible for rewards.
+
+If we cannot complete DKG successfully
+with at least _C_ participants
+within the allowed time,
+the failed wallet will not be used by TBTC.
+
+==== Group size and threshold
+
+Assuming we create groups every 24h and the lifespan of TBTC is 100 years,
+we would perform 36,500 sortitions in total.
+If an adversary has a 1:1,000,000 probability of controlling any given group,
+they'd have a roughly 3.6% probability of controlling one group at some point.
+
+Assuming 51/100 (N = 100, T = 51) would make an acceptably secure wallet,
+we can compare it to other parameter choices of similar security.
+
+It can be generally observed,
+that a slight increase in the T:N ratio
+compensates for a fairly large reduction in N (and T).
+A lower _N_ significantly reduces sortition costs.
+The downside of this is that the group has fewer redundant members
+which requires greater reliability from each individual node.
+In addition,
+a malicious actor with more than _N - T_ members
+is guaranteed to be able to DoS the group.
+In practice this is less significant
+as non-attributable threshold ECDSA protocols are vulnerable to DoS
+by an adversary controlling a significantly smaller number of members.
+
+From modeling we get the following security levels for different parameters
+against adversaries of various power:
+
+[%header,cols=7]
+|===
+|T/N |Bits against 0.15 |0.2 |0.25 |0.3 |0.35 |0.4
+
+|76/150 |>54 |>54 |36.74 |23.64 |14.23 |7.66
+
+|51/100 |>54 |38.00 |25.74 |16.91 |10.49 |5.94
+
+|41/80 |>54 |31.19 |21.36 |14.21 |8.97 |5.23
+
+|44/80 |>54 |37.90 |26.78 |18.56 |12.38 |7.78
+
+|36/70 |40.30 | 27.82 |19.16 |12.85 |8.21 |4.87
+
+|40/70 |>54 |37.01 |26.66 |18.93 |13.03 |8.54
+
+|26/50 |29.95 |21.07 |14.75 |10.11 |6.66 |4.14
+
+|32/50 |>54 |36.07 |27.29 |20.53 |15.20 |10.95
+|===
```

Comment:
Could be worth comparing estimated sortition cost cc @beaushinkle 

Off-chain, we have much more flexibility. We have not tried with `tss-lib` yet but for the random beacon GJKR and the same broadcast channel implementation, we were able to generate 128 member groups with no problems.

Reply:
I think sortition pools are fairly linear so if 150 members costs 3M gas, 100 members should be 2M gas, 70 members should be around 1.4M gas, and 50 members should be around 0.5M gas

Reply:
Ran the tests with 100 iterations against the simplest possible staking contract and got:

```
150 members
mean	3208155
stddev	31427

100 members
mean	2150957
stddev	43262

70 members
mean	1508201
stddev	15485

50 members
mean	1085133
stddev	2532
```
linearity: confirmed